pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -440,6 +536,10 @@ func (t *ResolvedPipelineTask) IsFinallySkipped(facts *PipelineRunFacts) TaskSki
 			skippingReason = v1beta1.MissingResultsSkip
 		case t.skipBecauseWhenExpressionsEvaluatedToFalse(facts):
 			skippingReason = v1beta1.WhenExpressionsSkip
+		case t.skipBecausePipelineRunPipelineTimeoutReached(facts):
+			skippingReason = v1beta1.PipelineTimedOutSkip
+		case t.skipBecausePipelineRunFinallyTimeoutReached(facts):
+			skippingReason = v1beta1.FinallyTimedOutSkip
```

Comment:
just noticing that stopping mode is not handled in `IsFinallySkipped` 🤔 

expected it to be here, or do we check if it's in stopping mode before starting finally tasks (which run in parallel)?

Reply:
I feel like we wouldn't want to be "stopping" if one of the finally tasks fails, since they're running in parallel...

Reply:
Oh, and `IsStopping` would return true if any of the non-finally tasks failed, so we definitely don't want to use that as a signal to not run finally tasks!